arangod/Aql/ExecutionBlock.cpp

```diff
@@ -51,138 +50,30 @@ static std::string const& stateToString(ExecutionState state) {
-  auto it = _exeNode->getRegisterPlan()->varInfo.find(id);
-
-  if (it != _exeNode->getRegisterPlan()->varInfo.end()) {
-    return (*it).second.registerId;
-  }
-  return ExecutionNode::MaxRegisterId;
-}
-
-RegisterId ExecutionBlock::getRegister(Variable const* variable) const {
-  TRI_ASSERT(variable != nullptr);
-  return getRegister(variable->id);
-}
+      _dependencyPos(_dependencies.end()),
+      _collector(&engine->itemBlockManager()) {}
 
-bool ExecutionBlock::removeDependency(ExecutionBlock* ep) {
-  auto it = _dependencies.begin();
-  while (it != _dependencies.end()) {
-    if (*it == ep) {
-      _dependencies.erase(it);
-      _dependencyPos = _dependencies.end();
-      return true;
-    }
-    ++it;
-  }
-  return false;
-}
+ExecutionBlock::~ExecutionBlock() = default;
 
-/// @brief whether or not the query was killed
-void ExecutionBlock::throwIfKilled() {
+void ExecutionBlock::throwIfKilled() { // TODO Scatter & DistributeExecutor still using
   if (_engine->getQuery()->killed()) {
     THROW_ARANGO_EXCEPTION(TRI_ERROR_QUERY_KILLED);
   }
 }
 
-std::pair<ExecutionState, arangodb::Result> ExecutionBlock::initializeCursor(InputAqlItemRow const& input) {
-  if (_dependencyPos == _dependencies.end()) {
-    // We need to start again.
-    _dependencyPos = _dependencies.begin();
-  }
-  for (; _dependencyPos != _dependencies.end(); ++_dependencyPos) {
-    auto res = (*_dependencyPos)->initializeCursor(input);
-    if (res.first == ExecutionState::WAITING || !res.second.ok()) {
-      // If we need to wait or get an error we return as is.
-      return res;
-    }
-  }
-
-  for (auto& it : _buffer) {
-    returnBlock(it);
-  }
-  _buffer.clear();
-
-  _done = false;
-  _upstreamState = ExecutionState::HASMORE;
-  _pos = 0;
-  _skipped = 0;
-  _collector.clear();
-
-  TRI_ASSERT(getHasMoreState() == ExecutionState::HASMORE);
-  TRI_ASSERT(_dependencyPos == _dependencies.end());
-  return {ExecutionState::DONE, TRI_ERROR_NO_ERROR};
-}
-
-/// @brief shutdown, will be called exactly once for the whole query
-std::pair<ExecutionState, Result> ExecutionBlock::shutdown(int errorCode) {
-  if (_dependencyPos == _dependencies.end()) {
-    _shutdownResult.reset(TRI_ERROR_NO_ERROR);
-    _dependencyPos = _dependencies.begin();
-  }
-
-  for (; _dependencyPos != _dependencies.end(); ++_dependencyPos) {
-    Result res;
-    ExecutionState state;
-    try {
-      std::tie(state, res) = (*_dependencyPos)->shutdown(errorCode);
-      if (state == ExecutionState::WAITING) {
-        return {state, TRI_ERROR_NO_ERROR};
-      }
-    } catch (...) {
-      _shutdownResult.reset(TRI_ERROR_INTERNAL);
-    }
-
-    if (res.fail()) {
-      _shutdownResult = res;
-    }
-  }
-
-  if (!_buffer.empty()) {
-    for (auto& it : _buffer) {
-      delete it;
-    }
-    _buffer.clear();
-  }
-
-  return {ExecutionState::DONE, _shutdownResult};
-}
-
 // Trace the start of a getSome call
 void ExecutionBlock::traceGetSomeBegin(size_t atMost) {
```

Comment:
Please move `traceGetSomeBegin` and `traceGetSomeEnd` into the header file, so it can be inlined.